diff -up mutt-1.5.21/init.c.gpgme-1.2.0 mutt-1.5.21/init.c
--- mutt-1.5.21/init.c.gpgme-1.2.0	2010-08-25 18:31:40.000000000 +0200
+++ mutt-1.5.21/init.c	2011-10-26 10:29:12.166967476 +0200
@@ -52,6 +52,10 @@
 #include <sys/wait.h>
 #include <sys/time.h>
 
+#if defined(CRYPT_BACKEND_GPGME)
+#include <gpgme.h>
+#endif
+
 #define CHECK_PAGER \
   if ((CurrentMenu == MENU_PAGER) && (idx >= 0) &&	\
 	    (MuttVars[idx].flags & R_RESORT)) \
@@ -3143,6 +3147,11 @@ void mutt_init (int skip_sys_rc, LIST *c
 
   mutt_read_histfile ();
 
+#ifdef CRYPT_BACKEND_GPGME
+  /* needed since version 1.2.0, ticket #3300 */
+  gpgme_check_version (NULL);
+#endif
+
 #if 0
   set_option (OPTWEED); /* turn weeding on by default */
 #endif
