--- a/globals.h 2009-08-25 21:08:52.000000000 +0200
+++ b/globals.h 2010-12-11 13:34:36.934682237 +0100
@@ -227,6 +227,9 @@
 WHERE char *PgpSignAs;
 WHERE short PgpTimeout;
 WHERE char *PgpEntryFormat;
+/* custom patch */
+WHERE char *PgpMimeSignatureFilename;
+WHERE char *PgpMimeSignatureDescription;
 WHERE char *PgpClearSignCommand;
 WHERE char *PgpDecodeCommand;
 WHERE char *PgpVerifyCommand;
--- a/init.h    2010-09-15 17:39:31.000000000 +0200
+++ b/init.h    2010-12-11 13:32:21.630299369 +0100
@@ -1884,6 +1884,21 @@
   ** (PGP only)
   **
   */
+
+  /* custom patch */
+  { "pgp_mime_signature_filename", DT_STR, R_NONE, UL &PgpMimeSignatureFilename, UL "signature.asc"},
+  /*
+  ** .pp
+  ** This option sets the filename used for signature parts in PGP/MIME
+  ** signed messages.
+  */
+  { "pgp_mime_signature_description", DT_STR, R_NONE, UL &PgpMimeSignatureDescription, UL "Digital signature"},
+  /*
+  ** .pp
+  ** This option sets the Content-Description used for signature parts in
+  ** PGP/MIME signed messages.
+  */
+
   { "pgp_retainable_sigs", DT_BOOL, R_NONE, OPTPGPRETAINABLESIG, 0 },
   /*
   ** .pp
--- a/pgp.c 2010-09-13 19:19:55.000000000 +0200
+++ b/pgp.c 2010-12-11 13:32:07.926705504 +0100
@@ -1130,6 +1130,11 @@
   t->encoding = ENC7BIT;
   t->unlink = 1; /* ok to remove this file after sending. */

+  /* custom patch */
+  mutt_set_parameter ("name", PgpMimeSignatureFilename, &t->parameter);
+  t->description = safe_strdup (PgpMimeSignatureDescription);
+
+
   return (a);
 }

