diff -ur profile-sync-daemon-3.10/profile-sync-daemon profile-sync-daemon-3.10.new/profile-sync-daemon
--- profile-sync-daemon-3.10/profile-sync-daemon	2012-08-11 23:29:06.000000000 +0200
+++ profile-sync-daemon-3.10.new/profile-sync-daemon	2012-08-24 14:13:32.911487944 +0200
@@ -18,7 +18,7 @@
 	exit 0
 fi
 
-BROWSERS=${BROWSERS:-"chromium google-chrome midori mozilla opera opera-next"} # all supported browsers
+BROWSERS=${BROWSERS:-"chromium google-chrome midori mozilla opera opera-next qupzilla"} # all supported browsers
 
 [[ -z "$VOLATILE" ]] && VOLATILE="/tmp"
 
@@ -44,7 +44,7 @@
 			BACKUP="$homedir/.config/$browser-backup"
 			DIR="$homedir/.config/$browser"
 			;;
-		mozilla|opera|opera-next)
+		mozilla|opera|opera-next|qupzilla)
 			BACKUP="$homedir/.$browser-backup"
 			DIR="$homedir/.$browser"
 			;;
