diff -ur profile-sync-daemon-3.5/profile-sync-daemon profile-sync-daemon-3.5.new/profile-sync-daemon
--- profile-sync-daemon-3.5/profile-sync-daemon	2012-07-04 18:29:28.744490042 +0200
+++ profile-sync-daemon-3.5.new/profile-sync-daemon	2012-07-04 18:30:17.346407941 +0200
@@ -2,6 +2,8 @@
 # By graysky <graysky AT archlinux DOT us>
 # Inspired by some code originally  written by Colin Verot
 
+PATH=/bin:/sbin:$PATH
+
 . /etc/psd.conf
 
 # nothing to do if there is no conf file
